py/desisurvey/afternoonplan.py

```diff
@@ -173,12 +173,10 @@ def afternoonPlan(self, day_stats, tiles_observed):
 
         # Copy the STATUS for previously observed tiles.
         if nto > 0:
-            joined = astropy.table.join(self.tiles, tiles_observed,
-                                        keys='TILEID', join_type='left',
-                                        table_names=['OLD', 'NEW'])
-            updated = ~joined['STATUS_NEW'].mask
-            sys.stdout.flush()
-            self.tiles['STATUS'][updated] = joined['STATUS_NEW'][updated]
+            for status in set(tiles_observed['STATUS']):
+                ii = (tiles_observed['STATUS'] == status)
+                jj = np.in1d(self.tiles['TILEID'], tiles_observed['TILEID'][ii])
+                self.tiles['STATUS'][jj] = status
 
```

Comment:
I didn't understand the original logic of this code (at most one night's worth of STATUS should have changed since the last call to `afternoonPlan`), but your implementation of it looks good.